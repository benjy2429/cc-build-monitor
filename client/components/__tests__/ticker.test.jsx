import React from 'react';
import { shallow, mount } from 'enzyme';
import Ticker from '../ticker';

describe('Ticker', () => {
  it('formats a duration in seconds', () => {
    const thirtySeconds = 30 * 1000;
    const component = shallow(<Ticker time={thirtySeconds} />);
    expect(component).toMatchSnapshot();
  });

  it('formats a duration in minutes', () => {
    const fiveMinutes = 5 * 60 * 1000;
    const component = shallow(<Ticker time={fiveMinutes} />);
    expect(component).toMatchSnapshot();
  });

  it('formats a duration in hours', () => {
    const oneHour = 60 * 60 * 1000;
    const component = shallow(<Ticker time={oneHour} />);
    expect(component).toMatchSnapshot();
  });

  it('defaults to 0 if no time is specified', () => {
    const component = shallow(<Ticker />);
    expect(component).toMatchSnapshot();
  });

  it('increments the time every second', () => {
    jest.useFakeTimers();
    const component = mount(<Ticker ticking />);
    expect(setInterval.mock.calls.length).toBe(1);
    expect(setInterval.mock.calls[0][1]).toBe(1000);

    jest.runOnlyPendingTimers();
    expect(component).toMatchSnapshot();
  });
});
