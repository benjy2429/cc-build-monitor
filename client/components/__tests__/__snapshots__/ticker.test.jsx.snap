// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker defaults to 0 if no time is specified 1`] = `
<time>
  00
</time>
`;

exports[`Ticker formats a duration in hours 1`] = `
<time>
  1h
</time>
`;

exports[`Ticker formats a duration in minutes 1`] = `
<time>
  05
</time>
`;

exports[`Ticker formats a duration in seconds 1`] = `
<time>
  30
</time>
`;

exports[`Ticker increments the time every second 1`] = `
<Ticker
  ticking={true}
>
  <time>
    01
  </time>
</Ticker>
`;
